activesupport/lib/active_support/core_ext/array/conversions.rb

```diff
@@ -50,35 +59,47 @@ class Array
   #   #         words_connector: " o "
   #   #         two_words_connector: " y "
   #   #         last_word_connector: " o al menos "
+  #   #         hidden_items_word: "otro"
   #
   #   ['uno', 'dos'].to_sentence(locale: :es)
   #   # => "uno y dos"
   #
   #   ['uno', 'dos', 'tres'].to_sentence(locale: :es)
   #   # => "uno o dos o al menos tres"
+  #
+  #   ['uno', 'dos', 'tres'].to_sentence(locale: :es, show_first: 1)
+  #   # => "uno y 2 otros"
   def to_sentence(options = {})
-    options.assert_valid_keys(:words_connector, :two_words_connector, :last_word_connector, :locale)
+    options.assert_valid_keys(:words_connector, :two_words_connector, :last_word_connector, :locale, :show_first, :hidden_items_word)
 
     default_connectors = {
       words_connector: ", ",
       two_words_connector: " and ",
-      last_word_connector: ", and "
+      last_word_connector: ", and ",
+      hidden_items_word: "other"
     }
     if defined?(I18n)
       i18n_connectors = I18n.translate(:'support.array', locale: options[:locale], default: {})
       default_connectors.merge!(i18n_connectors)
     end
     options = default_connectors.merge!(options)
 
-    case length
-    when 0
-      ""
-    when 1
-      "#{self[0]}"
-    when 2
-      "#{self[0]}#{options[:two_words_connector]}#{self[1]}"
+    show_first = options.delete(:show_first)
+    if show_first && length > show_first
+      hidden_items_length = length - show_first
+      hidden_items_suffix = "#{hidden_items_length} #{options[:hidden_items_word]}".pluralize(hidden_items_length)
+      (take(show_first) << hidden_items_suffix).to_sentence(options)
```

Comment:
I considered extracting some private functions, but I didn't think that would be a good idea since it would pollute the `Array` class more than is probably necessary.